cmd/metrics-v3-cluster-config.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-2024 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import "context"
+
+const (
+	configWriteQuorum    = "write_quorum"
+	configRRSParity      = "rrs_parity"
+	configStandardParity = "standard_parity"
+)
+
+var (
+	configWriteQuorumMD = NewGaugeMD(configWriteQuorum,
+		"Maximum write quorum across all pools and sets")
+	configRRSParityMD = NewGaugeMD(configRRSParity,
+		"Reduced redundancy storage class parity")
+	configStandardParityMD = NewGaugeMD(configStandardParity,
+		"Standard storage class parity")
+)
+
+// loadClusterConfigMetrics - `MetricsLoaderFn` for cluster config
+// such as standard and RRS parity.
+func loadClusterConfigMetrics(ctx context.Context, m MetricValues, c *metricsCache) error {
+	clusterDriveMetrics, _ := c.clusterDriveMetrics.Get()
+	m.Set(configStandardParity, float64(clusterDriveMetrics.storageInfo.Backend.StandardSCParity))
+	m.Set(configRRSParity, float64(clusterDriveMetrics.storageInfo.Backend.RRSCParity))
+
+	objLayer := newObjectLayerFn()
```

Comment:
``` suggestion
	objLayer := newObjectLayerFn()
```

This can be nil; pass this via `loader()` as an optional value.

Reply:
the function will not be called when objLayer is not initialized; so it cannot be nil here.

Reply:
Yeah but that is a very ugly style. In future when this code will move elsewhere it will crash.

Reply:
@harshavardhana I added the central checking because of this comment: https://github.com/minio/minio/pull/19507#discussion_r1580620169

> mainly look at metrics-v2.go object layer is checked centrally to avoid passing a nil before populating the values.

Did I misunderstand it? If so, and if that's not required, then the previous code should just work fine? something like

```
objLayer := newObjectLayerFn()
if objLayer != nil {
 // load metrics
}
```

Reply:
@harshavardhana I found that the the metric for which I was using `objLayer` (`write_quorum`) is already implemented under `/cluster/erasure-set` and would have become a duplicate here. So I have removed this metric (and other changes related to it) altogether from this PR.